### PR TITLE
BrAPI locations endpoints fixes

### PR DESCRIPTION
**Pull request by ctucker3**
Description 
-----------------------------------------------------
- Returns the new location on POST

I need to run the tests still. 

Adresses: https://github.com/solgenomics/sgn/issues/3529

The coordinate parsing turned out not to be a problem. 

Checklist 
---------------------------------------------------------------------------------------
- [ ] Refactoring only
- [ ] Documentation only
- [ ] Fixture update only
- [ x] Bug fix
  - [ ] The relevant issue has been closed.
  - [ ] Further work is required.
- [ ] New feature
  - [ ] Relevant tests have been created and run.
  - [ ] Data was added to the fixture
    - [ ] Data was added via a patch in `/t/data/fixture/patches/`.
  - [ ] User-Facing Change
    - [ ] The user manual in `/docs` has been updated.
  - [ ] Any new Perl has been documented using **perldoc**.
  - [ ] Any new JavaScript has been documented using **JSDoc**.
  - [ ] Any new _legacy_ JavaScript has been moved from `/js` to `/js/source/legacy`.

### CODE REVIEW

File: lib/CXGN/BrAPI/v2/Locations.pm [hunk 1]
```diff
@@ -174,11 +165,18 @@ sub detail {
 		};
 	}
 
-	my $result = $data;
 	my @data_files;
-	return CXGN::BrAPI::JSONResponse->return_success($result, $pagination, \@data_files, $status, 'Locations list result constructed');
+
+	if ($array) {
+		my %result = (data => \@data);
+		return CXGN::BrAPI::JSONResponse->return_success(\%result, $pagination, \@data_files, $status, 'Locations list result constructed');
```
Comment (MFlores2021) [inline lib/CXGN/BrAPI/v2/Locations.pm:172]: fix messages in the tests

File: lib/CXGN/BrAPI/v2/Locations.pm [hunk 2]
```diff
@@ -247,10 +245,10 @@ sub store {
...
 		}
 	}
-	my %result;
-	my $count = scalar @location_ids;
-    my $pagination = CXGN::BrAPI::Pagination->pagination_response($count,$page_size,$page);
-    return CXGN::BrAPI::JSONResponse->return_success( \%result, $pagination, undef, $self->status(), $count . " Locations were saved.");
+
+	# Retrieve our new locations
+	my $locations = CXGN::Trial::get_all_locations($schema, \@location_ids);
+	return $self->get_response($locations, 1);
```
Comment (MFlores2021) [inline lib/CXGN/BrAPI/v2/Locations.pm:251]: change tests

**Review by MFlores2021 — COMMENTED** [review]
Great. Just tests need to be fixed for the changes

File: lib/CXGN/BrAPI/v2/Locations.pm (hunk 1 above)
Comment (ctucker3) [reply lib/CXGN/BrAPI/v2/Locations.pm:172]: fixes tests relevant to locations

File: lib/CXGN/BrAPI/v2/Locations.pm (hunk 2 above)
Comment (ctucker3) [reply lib/CXGN/BrAPI/v2/Locations.pm:251]: fixes tests relevant to locations

**Review by MFlores2021 — APPROVED** [review]
:)

**Review by lukasmueller — APPROVED** [review]
:-)

**Pull request merged**